### DISCUSSION TITLE
New release for 2024?

### DISCUSSION THREAD

**Discussion by aburt2** [IDMIL/T-Stick#46] [General]
Once pull requests #45  and #44 are merged, I reckon its a good time for a new release. October 2021, was the last one and there has been a lot of changes since then. Also the puara gestures library is significantly changing soon so a new release before those changes would be appropriate.

New changes since October 2021:
- Switching to using Puara library #9 
- Improvements in OSC and libmapper code #44 and #45 
- Fixes in signal ranges in #27 
- IMU code refractoring #23 

If anyone remembers more feel free to add to the list. @edumeneses, @bboettcher3 , @malloch  what do you think?

**Comment by edumeneses** [comment]
On my end, there was nothing apart from what is already mentioned. @bboettcher3 might have something from SAT's Mitacs, although I don't remember touching the firmware at that point

**Reply by bboettcher3** [reply]
Nothing new to the firmware, though we did make https://github.com/libmapper/ForwardMapper around that time.. but not directly related to the T-Stick (unless the ESP32 issues are still there when multiple are connected at once)

**Reply by malloch** [reply]
How much testing has been done with this version (out of curiosity)? The problems with libmapper admin traffic which led Brady & Edu to create ForwardMapper hinted at very high packet loss – was this resolved?

**Reply by aburt2** [reply]
Everytime I've tested libmapper by itself on the ESP32 I haven't experienced high packet loss, but I know that calling lo_send 20-40 times per loop was starting to overwhelm the Wi-Fi driver on ESP32, and lead to both slow execution times (9-11ms) and slow sending rates (30-50Hz depending on the packet size)

**Reply by aburt2** [reply]
I haven't tested libmapper yet, with this firmware, I know that at least for OSC there is no significant packet loss

**Reply by malloch** [reply]
Thanks @aburt2 – that makes sense.

**Comment by malloch** [comment]
I suggest adding a configuration flag to disable sending OSC outside of libmapper. One of the points of libmapper from its earliest days was to throttle unused OSC traffic rather than spamming the network.

**Reply by aburt2** [reply]
Definitely, you can technically already do that by setting the IP address for OSC to either `0.0.0.0` or "", then  `puara.IP1/2ready()` will return false

**Reply by aburt2** [reply]
I was planning on changing the default config file to set the IP2 address to 0.0.0.0 so that the second address is disabled by default. Though maybe having them both be 0.0.0.0 by default would be better
